src/main/java/org/thingsboard/gateway/service/gateway/MqttGatewayClient.java

```diff
@@ -0,0 +1,580 @@
+import org.thingsboard.gateway.service.data.AttributeResponse;
+import org.thingsboard.gateway.service.data.RpcCommandData;
+import org.thingsboard.gateway.service.data.RpcCommandResponse;
+import org.thingsboard.gateway.util.JsonTools;
+import org.thingsboard.server.common.data.kv.BooleanDataEntry;
+import org.thingsboard.server.common.data.kv.DoubleDataEntry;
+import org.thingsboard.server.common.data.kv.KvEntry;
+import org.thingsboard.server.common.data.kv.LongDataEntry;
+import org.thingsboard.server.common.data.kv.StringDataEntry;
+import org.thingsboard.server.common.data.kv.TsKvEntry;
+
+import javax.annotation.PostConstruct;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyStore;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import static org.thingsboard.gateway.util.JsonTools.fromString;
+import static org.thingsboard.gateway.util.JsonTools.getKvEntries;
+import static org.thingsboard.gateway.util.JsonTools.newNode;
+import static org.thingsboard.gateway.util.JsonTools.putToNode;
+import static org.thingsboard.gateway.util.JsonTools.toBytes;
+
+/**
+ * Created by ashvayka on 16.01.17.
+ */
+@Slf4j
+public class MqttGatewayClient implements MqttHandler, MqttClientCallback {
+
+    private static final String GATEWAY_RPC_TOPIC = "v1/gateway/rpc";
+    private static final String GATEWAY_ATTRIBUTES_TOPIC = "v1/gateway/attributes";
+    private static final String GATEWAY_TELEMETRY_TOPIC = "v1/gateway/telemetry";
+    private static final String GATEWAY_REQUESTS_ATTRIBUTES_TOPIC = "v1/gateway/attributes/request";
+    private static final String GATEWAY_RESPONSES_ATTRIBUTES_TOPIC = "v1/gateway/attributes/response";
+    private static final String GATEWAY_CONNECT_TOPIC = "v1/gateway/connect";
+    private static final String GATEWAY_DISCONNECT_TOPIC = "v1/gateway/disconnect";
+    private static final String GATEWAY = "GATEWAY";
+
+    private static final String DEVICE_RPC_REQUEST_TOPIC = "v1/devices/me/rpc/request";
+    private static final String DEVICE_RPC_REQUEST_PLUS_TOPIC = "v1/devices/me/rpc/request/+";
+    private static final String DEVICE_RPC_RESPONSE_TOPIC = "v1/devices/me/rpc/response";
```

Comment:
```java
private static final String DEVICE_RPC_REQUEST_TOPIC = "v1/devices/me/rpc/request";
private static final String DEVICE_RPC_REQUEST_PLUS_TOPIC = "v1/devices/me/rpc/request/+";
private static final String DEVICE_RPC_RESPONSE_TOPIC = "v1/devices/me/rpc/response";
```
These are the topic subscriptions that are added. Are they intended to be used by Device-initiated RPC call? or Two-Way Server-Side RPC call? Also, there is no processing for these topics in onMessage() method